test/pom.xml

```diff
@@ -0,0 +1,172 @@
+<!--
```

Comment:
Just a question: If we are not going to let them run this test code, and this pr is to add naive implementation is it necessary to include pom.xml?

Reply:
Good catch! If @darrnshn thinks it's unnecessary too, I'll delete! The downside of taking it out is that if someone wants to run the tests themselves they have to generate the pom.xml themselves instead of just copying code files.